### PR TITLE
QA-192: Containerize tool and introduce Pipeline to the repo

### PR DESCRIPTION
**Pull request by lluiscampos**
Added the required files to build and run the tool through a Linux
container.

The docker-compose.yaml file is added aiming for local development while
the kubernetes.yaml file will describe the future deployment in the
production VM.

Added a pipeline to perform Go static checks and carry the build/publish
of the image.

### CODE REVIEW

**Comment by lluiscampos** [comment]
@0lmi I envision this PR as a "conversation starter" wrt how to do Continuous Deployment for Mender tools.

Please review specifically this `kubernetes.yaml` file intended to be applied on the cluster that will be hosting this tool. We can decide here what the convention should be for deployment or secrets names.

Your feedback is very much appreciated!

**Comment by oleorhagen** [comment]
I don't fully understand the motivation here. We already have just one environment where this needs to work right?

Could you expand on this a bit?

**Comment by lluiscampos** [comment]
> I don't fully understand the motivation here. We already have just one environment where this needs to work right?
> 
> Could you expand on this a bit?

@oleorhagen  Sorry, I thought you were in the loop.

Related to the certificates renewal of few days ago, we had a discussion and decided to stop the mender-jenkins VM (currently running on DO infra). To be able to do this, we have to move away the bot to another VM or to a kubernetes based cluster. SRE team has asked for us to containerize the tool so that we can get it up and running easier somewhere else.

**Comment by lluiscampos** [comment]
@oleorhagen Also please note that I have now pushed some minor code changes.

**Comment by oleorhagen** [comment]
> > I don't fully understand the motivation here. We already have just one environment where this needs to work right?
> > Could you expand on this a bit?
> 
> @oleorhagen Sorry, I thought you were in the loop.
> 
> Related to the certificates renewal of few days ago, we had a discussion and decided to stop the mender-jenkins VM (currently running on DO infra). To be able to do this, we have to move away the bot to another VM or to a kubernetes based cluster. SRE team has asked for us to containerize the tool so that we can get it up and running easier somewhere else.

Sense makin

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  - project: 'Northern.tech/Mender/mendertesting'
+    file: '.gitlab-ci-check-docker-build.yml'
+  - project: 'Northern.tech/Mender/mendertesting'
+    file: '.gitlab-ci-github-status-updates.yml'
+
+test:lint:kubeval:
+  stage: test
+  image: alpine
+  before_script:
+    - wget https://github.com/instrumenta/kubeval/releases/latest/download/kubeval-linux-amd64.tar.gz
```
Comment (oleorhagen) [inline .gitlab-ci.yml:21]: Strangely cumbersome install instruction for something I figured would be a pretty known tool?

File: main.go [hunk 2]
```diff
@@ -683,7 +683,7 @@ func syncIfOSHasEnterpriseRepo(conf *config, gpr *github.PullRequestEvent) error
 				return fmt.Errorf("syncIfOSHasEnterpriseRepo: Failed to create a PR with error: %v", err)
 			}
 
-			log.Infof("syncIfOSHasEnterpriseRepo: Created PR: %s on Enterprise/%s/%s",
+			log.Infof("syncIfOSHasEnterpriseRepo: Created PR: %d on Enterprise/%s/%s",
```
Comment (oleorhagen) [inline main.go:686]: Nice cleanups. Thx

File: main.go [hunk 3]
```diff
@@ -988,25 +988,25 @@ func getBuildParameters(conf *config, build *buildOptions) ([]*gitlab.PipelineVa
 				return nil, err
 			}
 			log.Infof("%s version %s is being used in %s", versionedRepo, version, build.baseBranch)
-			buildParameters = append(buildParameters, &gitlab.PipelineVariable{repoToBuildParameter(versionedRepo), version})
+			buildParameters = append(buildParameters, &gitlab.PipelineVariable{Key: repoToBuildParameter(versionedRepo), Value: version})
```
Comment (oleorhagen) [inline main.go:991]: Named parameters, yes please!

File: .gitlab-ci.yml (hunk 1 above)
Comment (lluiscampos) [reply .gitlab-ci.yml:21]: I just followed their [installation guide](https://www.kubeval.com/installation/)...

**Comment by 0lmi** [comment]
@lluiscampos We need way more yamls to make it working and it sounds to me like a separate task, so, I'm considering this as the template for future. I'm also not sure that this is the right place  for k8s files, we store all of them in separate repo, but again it can be here for now and move to proper place when we work on hosting it.

**Comment by lluiscampos** [comment]
@0lmi Got it. Thanks.

As I said above this PR is to let SRE define the "standard" on how to do Continuous Deployment. If you think a separate repo is best, then let's follow up with a JIRA task for it and move this yaml file as a "demo" for other tools to follow.

**Comment by lluiscampos** [comment]
@oleorhagen @0lmi Also, can one of you approve? Approval is required for merge.

**Comment by 0lmi** [comment]
@lluiscampos just thought about one more thing.  `image: integration-test-runner:master` and `imagePullPolicy: Always` doesn't allow to rollback. Probably this is not so important at this stage because `build` and `publish` phases are not fully implemented, but this is something we need to remember for future.

**Pull request merged**